### PR TITLE
Fix error message wording for PutObjectLockConfig

### PR DESCRIPTION
**Pull request by poornas**
to be S3 compatible.

## How to test this PR?
```
mc mb myminio/bucket
mc lock --clear myminio/bucket
```
Should show 
```
mc: <ERROR> Cannot enable object lock configuration on the specified bucket. Object Lock configuration cannot be enabled on existing buckets.
```
## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

**Comment by kannappanr** [comment]
This is the response from AWS
```
mc: <DEBUG> HTTP/1.1 409 Conflict
Connection: close
Transfer-Encoding: chunked
Content-Type: application/xml
Date: Tue, 07 Jan 2020 00:04:49 GMT
Server: AmazonS3
X-Amz-Id-2: xxxx
X-Amz-Request-Id: 20B3F76D6DCF136D

12b
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>InvalidBucketState</Code><Message>Object Lock configuration cannot be enabled on existing buckets</Message><RequestId>20B3F76D6DCF136D</RequestId><HostId>...</HostId></Error>
0

mc: <DEBUG> Response Time:  323.584935ms
```

File: cmd/api-errors.go [hunk 1]
```diff
@@ -732,6 +733,11 @@ var errorCodes = errorCodeMap{
 		Description:    "Bucket is missing ObjectLockConfiguration",
 		HTTPStatusCode: http.StatusBadRequest,
 	},
+	ErrObjectLockConfigurationNotAllowed: {
+		Code:           "InvalidRequest",
+		Description:    "Object Lock configuration cannot be enabled on existing buckets.",
+		HTTPStatusCode: http.StatusBadRequest,
```
Comment (kannappanr) [inline cmd/api-errors.go]:
```suggestion
		HTTPStatusCode: http.StatusConflict,
```

File: cmd/api-errors.go [hunk 2]
```diff
@@ -732,6 +733,11 @@ var errorCodes = errorCodeMap{
 		Description:    "Bucket is missing ObjectLockConfiguration",
 		HTTPStatusCode: http.StatusBadRequest,
 	},
+	ErrObjectLockConfigurationNotAllowed: {
+		Code:           "InvalidRequest",
```
Comment (kannappanr) [inline cmd/api-errors.go]:
```suggestion
		Code:           "InvalidBucketState",
```

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#2041dc96-0c6e-4a00-be74-6e4f03611a3a) |
| mint-large-bucket.sh | :x: [more...](#02ee3f51-a78d-4053-8013-51ef234b864e) |
<a name="2041dc96-0c6e-4a00-be74-6e4f03611a3a"></a>
### 8759-3b7b5b2/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.52:30523
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp c228a279ad94:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 14 seconds
{
  "name": "awscli",
  "duration": 12176,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (XMinioServerNotInitialized) when calling the CreateBucket operation (reached max retries: 4): Server not initialized, please try again."
}

Executed 0 out of 13 tests successfully.
```
<a name="02ee3f51-a78d-4053-8013-51ef234b864e"></a>
### 8759-3b7b5b2/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.52:30656
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 4e4df5e956fe:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... done in 2 minutes and 24 seconds
(2/13) Running aws-sdk-java tests ... done in 1 seconds
(3/13) Running aws-sdk-ruby tests ... done in 7 seconds
(4/13) Running mc tests ... done in 1 minutes and 10 seconds
(5/13) Running minio-go tests ... FAILED in 2 minutes and 21 seconds
{
  "args": {
    "destination": {},
    "source": [
      {
        "Headers": {
          "X-Amz-Copy-Source": [
            "minio-go-test-wv25fd5wxjhtnccg/srcObject"
          ],
          "X-Amz-Copy-Source-If-Match": [
            "6194e1aad31d414148ea33843e7394b9-1"
          ],
          "X-Amz-Copy-Source-Range": [
            "bytes=0-5242879"
          ]
        }
      },
      {
        "Headers": {
          "X-Amz-Copy-Source": [
            "minio-go-test-wv25fd5wxjhtnccg/srcObject"
          ],
          "X-Amz-Copy-Source-If-Match": [
            "6194e1aad31d414148ea33843e7394b9-1"
          ],
          "X-Amz-Copy-Source-Range": [
            "bytes=0-5242879"
          ]
        }
      }
    ]
  },
  "duration": 14994,
  "error": "The bucket you tried to delete is not empty",
  "function": "ComposeObject(destination, sources)",
  "message": "Cleanup failed",
  "name": "minio-go: testUserMetadataCopyingWrapper",
  "status": "FAIL"
}

Executed 4 out of 13 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#3a50001f-4e4a-4238-ac6f-c85513a32b05) |
| mint-large-bucket.sh | :x: [more...](#90ecfc17-05ee-4618-8de9-f81c92b6b5ff) |
| mint-dist-xl.sh | :x: [more...](#d600434c-100c-4e5c-a73f-68d0136d9b0d) |
<a name="3a50001f-4e4a-4238-ac6f-c85513a32b05"></a>
### 8759-4e63e4b/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.53:30540
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 971130a51c28:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 15 seconds
{
  "name": "awscli",
  "duration": 12638,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (XMinioServerNotInitialized) when calling the CreateBucket operation (reached max retries: 4): Server not initialized, please try again."
}

Executed 0 out of 13 tests successfully.
```
<a name="d600434c-100c-4e5c-a73f-68d0136d9b0d"></a>
### 8759-4e63e4b/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:30709
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 971a33fa946c:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 5 minutes and 13 seconds
{"name": "awscli", "duration": 310972, "function": "make_bucket\n", "status": "FAIL", "error": "Connect timeout on endpoint URL: "http://72.28.97.55:30709/awscli-mint-test-bucket-19201""}

Executed 0 out of 13 tests successfully.
```
<a name="90ecfc17-05ee-4618-8de9-f81c92b6b5ff"></a>
### 8759-4e63e4b/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:31575
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp d392241dae86:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... done in 2 minutes and 25 seconds
(2/13) Running aws-sdk-java tests ... done in 1 seconds
(3/13) Running aws-sdk-ruby tests ... done in 8 seconds
(4/13) Running mc tests ... done in 1 minutes and 10 seconds
(5/13) Running minio-go tests ... FAILED in 3 minutes and 10 seconds
{
  "args": {},
  "duration": 80862,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 4 out of 13 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :x: [more...](#9eebc803-e6da-4955-ae1d-9ec31681d35f) |
| mint-large-bucket.sh | :x: [more...](#2b095d68-eb84-40eb-867d-cd40937a630a) |
| mint-compression-dist-xl.sh | :x: [more...](#035a4d33-a57e-4541-be06-a386d0787627) |
<a name="2b095d68-eb84-40eb-867d-cd40937a630a"></a>
### 8759-ca1f7c9/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:32741
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp e943a00cfaf6:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 5 minutes and 15 seconds
{"name": "awscli", "duration": 313115, "function": "make_bucket\n", "status": "FAIL", "error": "Connect timeout on endpoint URL: "http://72.28.97.55:32741/awscli-mint-test-bucket-3976""}

Executed 0 out of 13 tests successfully.
```
<a name="035a4d33-a57e-4541-be06-a386d0787627"></a>
### 8759-ca1f7c9/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:30541
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 277980980f8e:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 5 minutes and 13 seconds
{"name": "awscli", "duration": 310912, "function": "make_bucket\n", "status": "FAIL", "error": "Connect timeout on endpoint URL: "http://72.28.97.55:30541/awscli-mint-test-bucket-17741""}

Executed 0 out of 13 tests successfully.
```
<a name="9eebc803-e6da-4955-ae1d-9ec31681d35f"></a>
### 8759-ca1f7c9/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.58:30489
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp f0a3a76b52d2:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 11 seconds
{
  "name": "awscli",
  "duration": 8285,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (XMinioServerNotInitialized) when calling the CreateBucket operation (reached max retries: 4): Server not initialized, please try again."
}

Executed 0 out of 13 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Review by kannappanr — APPROVED** [review]
LGTM

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**